cudamapper/src/overlapper.cpp

```diff
@@ -204,5 +244,90 @@ void Overlapper::print_paf(const std::vector<Overlap>& overlaps, const std::vect
         idx++;
     }
 }
+
+void Overlapper::post_process_overlaps(std::vector<Overlap>& overlaps)
+{
+    const auto num_overlaps = overlaps.size();
+    bool in_fuse            = false;
+    int fused_target_start;
+    int fused_query_start;
+    int fused_target_end;
+    int fused_query_end;
+    int num_residues = 0;
+    Overlap prev_overlap;
+
+    for (int i = 1; i < num_overlaps; i++)
+    {
+        prev_overlap                  = overlaps[i - 1];
+        const Overlap current_overlap = overlaps[i];
+        //Check if previous overlap can be merged into the current one
+        if (overlaps_mergable(prev_overlap, current_overlap))
+        {
+            if (!in_fuse)
+            { // Entering a new fuse
+                num_residues      = prev_overlap.num_residues_ + current_overlap.num_residues_;
+                in_fuse           = true;
+                fused_query_start = prev_overlap.query_start_position_in_read_;
+                fused_query_end   = current_overlap.query_end_position_in_read_;
+
+                // If the relative strands are forward, then the target positions are increasing.
+                // However, if the relative strands are in the reverse direction, the target
+                // positions along the read are decreasing. When fusing, this needs to be accounted for
+                // by the following checks.
+                if (current_overlap.relative_strand == RelativeStrand::Forward)
+                {
+                    fused_target_start = prev_overlap.target_start_position_in_read_;
+                    fused_target_end   = current_overlap.target_end_position_in_read_;
+                }
+                else
+                {
+                    fused_target_start = current_overlap.target_start_position_in_read_;
+                    fused_target_end   = prev_overlap.target_end_position_in_read_;
+                }
+            }
+            else
+            {
+                // Continuing a fuse, query end is always incremented, however whether we increment the target start or
+                // end depends on whether the overlap is a reverse or forward strand overlap.
+                num_residues += current_overlap.num_residues_;
+                fused_query_end = current_overlap.query_end_position_in_read_;
```

Comment:
No, both the target and the query need to advance. For the query, `fused_query_end`  will always advance since the overlapper guarantees that queries are returned in increasing order.

However, the target overlaps may be increasing or decreasing, depending on whether they are reverse or forward, hence we need the if block to decide whether to advance the end of the target (for forward strands) or the start (for reverse strands).

I have added a further comment to clarify.

Reply:
Ah no, that's ok. I didn't think it through. Good to leave it as it is.